simulator/source/Combat_simulator.cpp

```diff
@@ -1911,19 +1766,31 @@ void Combat_simulator::simulate(const Character& character, int init_iteration,
                     {
                         if (config.combat.use_heroic_strike)
                         {
-                            if (config.dpr_settings.compute_dpr_hs_)
-                            {
-                                dpr_heroic_strike_queued_ = true;
-                            }
-                            else
-                            {
-                                ability_queue_manager.queue_heroic_strike();
-                                simulator_cout("Heroic strike activated");
-                            }
+                            ability_queue_manager.queue_heroic_strike();
+                            simulator_cout("Heroic strike activated");
                         }
                     }
                 }
             }
+
+            // end of turn - update swing timers if necessary
+            if (mh_swing)
+            {
+                weapons[0].internal_swing_timer = weapons[0].swing_speed / (1 + special_stats.haste);
+            }
+            else if (special_stats.haste != oldHaste)
+            {
+                weapons[0].internal_swing_timer *= (1 + oldHaste) / (1 + special_stats.haste);
+            }
+
+            if (oh_swing)
+            {
+                weapons[1].internal_swing_timer = weapons[1].swing_speed / (1 + special_stats.haste);
+            }
+            else if (is_dual_wield && special_stats.haste != oldHaste)
+            {
+                weapons[1].internal_swing_timer *= (1 + oldHaste) / (1 + special_stats.haste);
+            }
```

Comment:
speed up or slow down remaining swing timers, or reset with updated haste. this is necessary for flurry, but also for any other haste proc.